app/models/queen.rb

```diff
@@ -1,2 +1,25 @@
 class Queen < Piece
+  def move_legal?(to_row, to_col)
+    row_diff = (row - to_row).abs
+    col_diff = (col - to_col).abs
+    row_diff <= 1 && col_diff <= 1
+  end
+
+  def valid_move?(row, col)
```

Comment:
Its better to let Piece's `valid_move?` do its thing, and move your own code to differently named methods.